java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertSwitchToRuleSwitch.java

```diff
@@ -0,0 +1,258 @@
+                scan(node.getElseStatement(), p);
+                completesNormally |= afterThen;
+                return null;
+            }
+            @Override
+            public Void visitSwitch(SwitchTree node, Void p) {
+                //exhaustiveness: (TODO)
+                boolean hasDefault = node.getCases().stream().anyMatch(c -> c.getExpression() == null);
+                if (node.getCases().size() > 0)
+                    scan(node.getCases().get(node.getCases().size() - 1), p);
+                completesNormally |= !hasDefault;
+                return null;
+            }
+            //TODO: loops
+            @Override
+            public Void visitLambdaExpression(LambdaExpressionTree node, Void p) {
+                return null;
+            }
+            @Override
+            public Void visitClass(ClassTree node, Void p) {
+                return null;
+            }
+        }
+
+        Scanner scanner = new Scanner();
+
+        scanner.scan(tp.getLeaf(), null);
+        return scanner.completesNormally;
+    }
+
+    private static final class FixImpl extends JavaFix {
+
+        public FixImpl(CompilationInfo info, TreePath switchStatement) {
+            super(info, switchStatement);
+        }
+
+        @Override
+        @Messages("FIX_ConvertToRuleSwitch=Convert to rule switch")
+        protected String getText() {
+            return Bundle.FIX_ConvertToRuleSwitch();
+        }
+
+        @Override
+        protected void performRewrite(TransformationContext ctx) {
+            WorkingCopy wc = ctx.getWorkingCopy();
+            TreePath tp = ctx.getPath();
+            TreeMaker make = wc.getTreeMaker();
+            SwitchTree st = (SwitchTree) tp.getLeaf();
+            List<CaseTree> newCases = new ArrayList<>();
+            List<ExpressionTree> patterns = new ArrayList<>();
+            Set<VariableElement> variablesDeclaredInOtherCases = new HashSet<>();
+
+            for (Iterator<? extends CaseTree> it = st.getCases().iterator(); it.hasNext();) {
+                CaseTree ct = it.next();
+                TreePath casePath = new TreePath(tp, ct);
+                patterns.addAll(TreeShims.getExpressions(ct));
+                List<StatementTree> statements = new ArrayList<>(ct.getStatements());
+                if (statements.isEmpty()) {
+                    if (it.hasNext())
+                        continue;
+                    //last case, no break
+                } else if (statements.get(statements.size() - 1).getKind() == Kind.BREAK &&
+                    ctx.getWorkingCopy().getTreeUtilities().getBreakContinueTarget(new TreePath(new TreePath(tp, ct), statements.get(statements.size() - 1))) == st) {
+                    statements.remove(statements.size() - 1);
+                } else {
+                    new TreePathScanner<Void, Void>() {
+                        @Override
+                        public Void visitBlock(BlockTree node, Void p) {
```

Comment:
Here, I opted for simplicity - it is not a compile-time error to have breaks in the -> switch/case, so while this may be producing a slightly non-optimal code in some (presumably) corner cases, that feels like something that can improve over time if needed.